rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -254,22 +254,40 @@ def decode_transaction_hashes(self, ignore_cache: bool, tx_hashes: List[EVMTxHas
         self.reload_from_db()
         tx_module = EthTransactions(ethereum=self.ethereum_manager, database=self.database)
 
-        for tx_hash in tx_hashes:
-            try:
-                receipt = tx_module.get_or_query_transaction_receipt(tx_hash)
-            except RemoteError as e:
-                raise InputError(f'Hash {tx_hash.hex()} does not correspond to a transaction') from e  # noqa: E501
-
-            # TODO: Change this if transaction filter query can accept multiple hashes
-            txs = self.dbethtx.get_ethereum_transactions(
-                filter_=ETHTransactionsFilterQuery.make(tx_hash=tx_hash),
-                has_premium=True,  # ignore limiting here
+        # If no transaction hashes are passed, decode all transactions.
+        if len(tx_hashes) == 0:
+            txns = self.dbethtx.get_ethereum_transactions(
```

Comment:
Hmm .. no no no no. What you should do and query in this case is just the transaction hashes. Not all transactions and then decode for each one. 

So you would populate a new tx_hashes list and then let it proceed down to the logic of decoding as it used to be. The only thing you need to do is if the transaction list is not given (is None) then create a new tx list with all hashes.